src/cmd.js

```diff
@@ -258,6 +288,46 @@ function action(actionToDispatch) {
   });
 }
 
+function schedule(nestedCmd, delayMs, onScheduledActionCreator) {
+  return _schedule(nestedCmd, delayMs, onScheduledActionCreator, false);
+}
+
+function scheduleRepeating(nestedCmd, delayMs, onScheduledActionCreator) {
+  return _schedule(nestedCmd, delayMs, onScheduledActionCreator, true);
+}
+
+function _schedule(nestedCmd, delayMs, onScheduledActionCreator, isRepeating) {
+  if (process.env.NODE_ENV !== 'production') {
+    throwInvariant(
+      isCmd(nestedCmd),
+      'Cmd.schedule: first argument must be another Cmd'
+    );
+    throwInvariant(
+      typeof delayMs === 'number',
+      'Cmd.schedule: second argument must be a number'
+    );
+    throwInvariant(
+      onScheduledActionCreator === undefined ||
+        typeof onScheduledActionCreator === 'function',
+      'Cmd.schedule: third argument must be a function (if specified)'
+    );
+  }
+
+  return Object.freeze({
+    [isCmdSymbol]: true,
+    type: cmdTypes.SCHEDULE,
+    nestedCmd,
+    delayMs,
+    isRepeating,
+    onScheduledActionCreator,
+    simulate: simulateSchedule
+  });
+}
+
+function simulateSchedule(simulation) {
+  return this.nestedCmd.simulate(simulation);
```

Comment:
hmmm we should do more than simulate the nested command. We want to be able to test the action creator too. 

i think the passed in simulation object should contain something like interval/timeout id, as  well as a nested cmd simulation object, and then the result should contain the nested result and the action that would get dispatched with the id.

something like this

```js
  return {
     nestedResult: this.nestedCmd.simulate(simulation.nestedSimulation),
     action: this.onScheduledActionCreator(simulation.timeoutId)
  };
```

Reply:
Simulate has to return `Array<Action>` or `Action` or `null`, otherwise it will break if nested inside another command like Cmd.list. Probably we should simplify this and have all simulate methods return an `Array<Action>` but this would be a breaking change so let's leave it for later.

Reply:
Now is the time to do breaking changes. The next release is going to have breaking changes in it anyway. And the action creator is the main thing simulate is designed to test, so I think it'd be good to include it.

That being said, we can probably avoid the breaking changes anyway. I didn't realize it would break the list and map simulations. What if we returned an array instead. 

```js
return this.nestedCmd.simulate(simulation.nestedSimulation).concat(this.onScheduledActionCreator(simulation.timeoutId))
```

Reply:
wait i see you already did that. perfect